services/npm/npm-unpacked-size.tester.js

```diff
@@ -0,0 +1,16 @@
+import { isFileSize } from '../test-validators.js'
+import { createServiceTester } from '../tester.js'
+
+export const t = await createServiceTester()
+
+t.create('Latest unpacked size')
+  .get('/express.json')
+  .expectBadge({ label: 'unpacked size', message: isFileSize })
+
+t.create('Nonexistent unpacked size with version')
+  .get('/express/4.16.0.json')
+  .expectBadge({ label: 'unpacked size', message: 'unknown' })
```

Comment:
After adding `schema` to validate NPM registry response body, this test fails saying this:

```plain
5) NpmUnpackedSize
       [live] Nonexistent unpacked size with version
         
        [ GET /express/4.16.0.json ]:

      label mismatch
      + expected - actual

      -size
      +unpacked size
```

When I manually requested [the local dev endpoint](http://localhost:8080/npm/unpacked-size/express/4.16.0), I've seen the badge as such:

![the badge with error](https://github.com/badges/shields/assets/2399084/d7081260-51ec-4b98-9eda-5937b91cc15a)

(could not paste the PNG of it as it's not available in the dev server)

I expected the label to be `unpacked size` instead of only `size`. It seems like, when the schema validation fails, it renders the label as `size`, which, I think, is a built-in mechanism.

I think when `dist.unpackedSize` is not defined, the label should still be `unpacked size` to imply that this is not any kind of other size.

@chris48s, do you have any idea as to how to override the default badge when `dist.unpackedSize` does not exist? I've tinkered a little bit with the codebase but I could not see any way (maybe I've missed it or haven't realized).

Reply:
Actually, I've checked it through `BaseJsonService._requestJson` method, which calls `BaseService._validate`. `BaseService._validate` has positional `options` parameter, in which, `prettyErrorMessage` has been set to `"invalid response data"`.

All the way up the stack (from `NpmUnpackedSize` service), I cannot set `prettyErrorMessage` because it is internal to `BaseJsonService` and `BaseService`. Given that `BaseJsonService._requestJson` and `BaseService._validate` are internals (prepended by underscore), I naturally hesitate to change them.

Thus, there are two possible paths here:

1. Modify `BaseService._validate` and `BaseJsonService._requestJson` methods so that I can reach and modify the message (and thus, the label) when `dist.unpackedSize` does not exist. Given that these methods are defined as internals and tons of services rely on their signature, this might/would cause enormous amount of failures on some tests in the best case, or unexpected behavior on runtime at worst.
2. Accept the sweet fate and let the badge stay as `size: invalid response data`. I thought it wouldn't be a viable solution as (given that JS is used in web) it is important to differ unpacked size from minified or min+gzipped sizes but there's no sane room here. So, I can only modify the test to satisfy `size: invalid response data` badge.

@chris48s What's your idea?

Reply:
Ah right - sorry. In my original post, the suggestions should have been

`optionalNonNegativeInteger` and `nonNegativeInteger`
not
`nonNegativeInteger` and `nonNegativeInteger.required()`

:facepalm: 

if we switch to using `optionalNonNegativeInteger` for the `unpackedSize` key, then you should be able to carry on doing what you were doing and render "unknown" if the `unpackedSize` key is missing.

The validation is still useful, because it ensures the response has a `dist` object and that `unpackedSize` is of the right type if it exists.

If you want to set the label as "unpacked size" even when an exception is thrown, setting

```js
static defaultBadgeData = { label: 'unpacked size' }
```

will do the job.

Does that unstick you?

Reply:
Thanks, did it. We definitely need documentation about these premade validators, missed it somehow. That's maybe the topic of another PR.

With that in mind, seems like this is completed.

Reply:
The validators are included in the internal api docs here: https://contributing.shields.io/module-services_validators.html